app/assets/javascripts/admin/addon/components/admin-page-header.gjs

```diff
@@ -39,87 +39,95 @@ export default class AdminPageHeader extends Component {
     }
   }
 
+  get shouldDisplay() {
+    return this.args.shouldDisplay === undefined
```

Comment:
I _think_ you can do this:

```suggestion
    return this.args.shouldDisplay ?? true;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
I wonder if we could rely on our `router` service here instead of having to manually define which routes to hide this on in each controller, which I think would be easy to forget. We want to make it as smooth as possible.

Take this for example:

https://github.com/discourse/discourse/blob/955884df0e1559bfacf5f2e8b395f3cbf307e731/app/assets/javascripts/admin/addon/controllers/admin.js#L25-L46

I think we could have logic to say "if the final route segment is `new` or `edit`, and we are in the `admin` route namespace, then do not display the header", but still allow overriding this with `this.args.shouldDisplay`. WDYT?

Reply:
There may just be some places where we need to update routes to be `edit` instead of show, which we needed to do anyway